app/file-detail/controller.js

```diff
@@ -102,39 +104,36 @@ export default Controller.extend({
             this.set('deleteModalOpen', false);
         },
 
-        changeViewPanel(panel, button) {
-            if (!this.get('isEditableFile')) {
-                $('#mainViewBtn, #revisionBtn').toggleClass('btn-primary btn-default');
-                $('#mfrIframeParent, #revisionsPanel').toggle();
-                return;
+        revisionBtnClick() {
+            this.set('show', 'revision');
+        },
+
+        editBtnClick() {
+            if (this.get('show') === 'view') {
+                this.set('show', 'view_edit');
+            } else if (this.get('show') === 'view_edit') {
+                this.set('show', 'view');
+            } else if (this.get('show') === 'revision') {
+                this.set('show', 'edit');
             }
+        },
 
-            if (button === 'revisionBtn') {
-                if ($(`#${panel}`).css('display') === 'none') {
-                    $('.panel-view').hide().removeClass('col-sm-6');
-                    $('.view-button').removeClass('btn-primary').addClass('btn-default');
-                } else {
-                    $('#mfrIframeParent').toggle();
-                    $('#mainViewBtn').toggleClass('btn-default btn-primary');
-                }
-                $(`#${panel}`).toggle();
-                $(`#${button}`).toggleClass('btn-default btn-primary');
-                return;
+        viewBtnClick() {
+            if (this.get('show') === 'edit') {
+                this.set('show', 'view_edit');
+            } else if (this.get('show') === 'view_edit') {
+                this.set('show', 'edit');
+            } else if (this.get('show') === 'revision') {
+                this.set('show', 'view');
             }
+        },
 
-            if ($(`#${button}`).hasClass('btn-primary') && $('.view-button.btn-primary').length === 1) {
-                return;
-            } else if ($('#revisionsPanel').css('display') !== 'none') {
-                $('#revisionsPanel').toggle();
-                $('#revisionBtn').toggleClass('btn-default btn-primary');
-            } else if ($('#mfrIframeParent').css('display') !== 'none' || $('#editPanel').css('display') !== 'none') {
-                $('.panel-view').toggleClass('col-sm-6');
+        versionLinkToggle() {
+            if (this.get('show') === 'revision') {
+                this.set('show', 'view');
             } else {
-                $('.panel-view').removeClass('col-sm-6');
+                this.set('show', 'revision');
             }
-
-            $(`#${panel}`).toggle();
-            $(`#${button}`).toggleClass('btn-default btn-primary');
         },
```

Comment:
This could be handled by the btnClick action suggested above (same rules as revision button).